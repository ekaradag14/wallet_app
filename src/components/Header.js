import React from "react";
import { NavLink } from "react-router-dom";
export default function Header() {
  return (
    <header>
      <h1>Expensify</h1>
      {/* ActiveClassName para cambiar css clase de pagina corriente */}
      <NavLink activeClassName="is-active-nav-link" exact={true} to="/">
        Dashboard
      </NavLink>
      <NavLink activeClassName="is-active-nav-link" to="/create">
        Create
      </NavLink>
      <NavLink activeClassName="is-active-nav-link" to="/edit">
        Edit
      </NavLink>
      <NavLink activeClassName="is-active-nav-link" to="/help">
        Help
      </NavLink>
    </header>
  );
}
