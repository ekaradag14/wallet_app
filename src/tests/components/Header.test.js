import React from 'react'
import { shallow } from 'enzyme'// //Shallow es solo para trabajar con componente mas bien que todo objeto de DOM.
import toJSON from 'enzyme-to-json'//Es paquete para cambiar el formato de nuestra snapshot para ser mas limpiar cuando usamos enzymeö
import Header from '../../components/Header'

describe('Header Component Testing Scenarios', ()=> {

    test('should render Header correctly', () => {
      const wrapper = shallow(<Header />);
      expect(toJSON(wrapper)).toMatchSnapshot(); //Creamos una copia de nuestra componente para vigilar. Si hay una diferencia que queremos guardar, vamos a pulsar u en command line
     //Normalmente no deberíamos usar toJSON porque tenemos     "snapshotSerializers":[  "enzyme-to-json/serializer"] en jest.config.json. Y ese hace este automáticamente pero usamos un Adapter no oficial. Probablemente tenemos un error allí.
 
    })
})
