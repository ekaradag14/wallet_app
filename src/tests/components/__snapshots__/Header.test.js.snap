// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Testing Scenarios should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active-nav-link"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active-nav-link"
    to="/create"
  >
    Create
  </NavLink>
  <NavLink
    activeClassName="is-active-nav-link"
    to="/edit"
  >
    Edit
  </NavLink>
  <NavLink
    activeClassName="is-active-nav-link"
    to="/help"
  >
    Help
  </NavLink>
</header>
`;
